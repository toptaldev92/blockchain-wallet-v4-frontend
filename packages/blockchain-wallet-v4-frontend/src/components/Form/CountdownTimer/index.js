import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import CountdownTimer from './template'

class CountdownTimerContainer extends React.PureComponent {
  constructor (props) {
    super(props)
    this.interval = undefined
    const { expiryDate } = this.props
    this.state = { expiration: moment(expiryDate), elapsed: moment.duration(moment(expiryDate).diff(moment())) }
    this.tick = this.tick.bind(this)
  }

  componentDidMount () {
    this.interval = setInterval(this.tick, 1000)
  }

  componentWillUnmount () {
    clearInterval(this.interval)
  }

  tick () {
    const { handleExpiry } = this.props
    const elapsed = moment.duration(moment(this.state.expiration).diff(moment()))
    if (this.state.elapsed.as('milliseconds') < 1) {
      // If we reach the end of the timer, we execute the expiry callback
      if (handleExpiry) { handleExpiry() }
    } else {
      // We increment the time elapsed
      this.setState({ elapsed })
    }
  }

  render () {
    const timeLeft = moment.utc(this.state.elapsed.as('milliseconds')).format('mm:ss')

    return (
      <CountdownTimer {...this.props} timeLeft={timeLeft} />
    )
  }
}
CountdownTimerContainer.propTypes = {
  expiryDate: PropTypes.number,
  handleExpiry: PropTypes.func
}

CountdownTimerContainer.defaultProps = {
  expiryDate: moment().add('minutes', 1)
}

export default CountdownTimerContainer
