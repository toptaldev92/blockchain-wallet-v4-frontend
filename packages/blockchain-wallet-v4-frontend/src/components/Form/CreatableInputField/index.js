import React from 'react'
import styled from 'styled-components'
import { CreatableInput } from 'blockchain-info-components'

const Container = styled.div`
  position: relative;
  width: 100%;
  height: auto;
`
const Error = styled.label`
  position: absolute;
  top: ${props => (props.errorBottom ? '40px' : '-20px')};
  right: 0;
  display: block;
  height: 15px;
  font-size: 12px;
  font-weight: 300;
  font-family: 'Montserrat', Helvetica, sans-serif;
  color: ${props => props.theme['error']};
`

const CreatableInputField = props => {
  const { input, meta, hideErrors, errorBottom, ...rest } = props
  const { touched, invalid, error, pristine } = meta
  const errorState = touched && invalid ? 'invalid' : 'initial'

  return (
    <Container>
      <CreatableInput {...input} {...meta} {...rest} errorState={errorState} />
      {(touched || !pristine) && error && !hideErrors && (
        <Error errorBottom={errorBottom}>{error}</Error>
      )}
    </Container>
  )
}

export default CreatableInputField
