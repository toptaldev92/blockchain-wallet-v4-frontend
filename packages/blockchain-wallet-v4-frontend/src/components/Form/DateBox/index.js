import React from 'react'
import styled from 'styled-components'

import { Text, DateInput } from 'blockchain-info-components'

const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  width: 100%;
  height: 40px;
`
const Error = styled(Text)`
  position: absolute;
  display: block;
  top: -18px;
  right: 0;
  height: 15px;
`
const getErrorState = (meta) => {
  return !meta.touched ? 'initial' : (meta.invalid ? 'invalid' : 'valid')
}

const DateBox = (field) => {
  const errorState = getErrorState(field.meta)

  return (
    <Container>
      <DateInput {...field.input} errorState={errorState} placeholder={field.placeholder} />
      {field.meta.touched && field.meta.error && <Error size='12px' weight={300} color='error'>{field.meta.error}</Error>}
    </Container>
  )
}

export default DateBox
