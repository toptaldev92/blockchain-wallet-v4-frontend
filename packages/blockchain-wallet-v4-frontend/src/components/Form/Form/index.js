import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Form as ReduxForm } from 'redux-form'

const BaseForm = styled(ReduxForm)`
  width: 100%;
`

const Form = props => {
  const { children, ...rest } = props

  return (
    <BaseForm {...rest}>
      {children}
    </BaseForm>
  )
}

export default Form
