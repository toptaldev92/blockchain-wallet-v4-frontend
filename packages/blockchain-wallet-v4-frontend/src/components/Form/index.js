
import CaptchaBox from './CaptchaBox'
import CheckBox from './CheckBox'
import CodeInputBoxes from './CodeInputBoxes'
import CountdownTimer from './CountdownTimer'
import DateBox from './DateBox'
import FiatConvertor from './FiatConvertor'
import Form from './Form'
import FormGroup from './FormGroup'
import FormItem from './FormItem'
import NumberBox from './NumberBox'
import PasswordBox from './PasswordBox'
import PlaceHolder from './PlaceHolder'
import PhoneNumberBox from './PhoneNumberBox'
import RadioButton from './RadioButton'
import SelectBox from './SelectBox'
import SelectBoxBitcoinAddresses from './SelectBoxBitcoinAddresses'
import SelectBoxBitcoinUnit from './SelectBoxBitcoinUnit'
import SelectBoxCoin from './SelectBoxCoin'
import SelectBoxCurrency from './SelectBoxCurrency'
import SelectBoxFee from './SelectBoxFee'
import SelectBoxLanguages from './SelectBoxLanguages'
import SelectBoxTheme from './SelectBoxTheme'
import SelectBoxUSState from './SelectBoxUSState'
import TabMenuAddressesStatus from './TabMenuAddressesStatus'
import TabMenuBuySellStatus from './TabMenuBuySellStatus'
import TabMenuTransactionStatus from './TabMenuTransactionStatus'
import TextArea from './TextArea'
import TextBox from './TextBox'

export {
  CaptchaBox,
  CheckBox,
  CodeInputBoxes,
  CountdownTimer,
  DateBox,
  FiatConvertor,
  Form,
  FormGroup,
  FormItem,
  NumberBox,
  PasswordBox,
  PlaceHolder,
  PhoneNumberBox,
  RadioButton,
  SelectBox,
  SelectBoxBitcoinAddresses,
  SelectBoxBitcoinUnit,
  SelectBoxCoin,
  SelectBoxCurrency,
  SelectBoxFee,
  SelectBoxLanguages,
  SelectBoxTheme,
  SelectBoxUSState,
  TabMenuAddressesStatus,
  TabMenuBuySellStatus,
  TabMenuTransactionStatus,
  TextArea,
  TextBox
}
