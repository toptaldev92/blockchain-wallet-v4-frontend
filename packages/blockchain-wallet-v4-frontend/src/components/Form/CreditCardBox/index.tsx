import { CommonFieldProps, WrappedFieldMetaProps } from 'redux-form'
import { DEFAULT_CARD_FORMAT, getCardTypeByValue } from './model'
import { FormattedMessage } from 'react-intl'
import { TextBox } from 'components/Form'
import React from 'react'

export const normalizeCreditCard = (value, previousValue) => {
  if (!value) return value

  const { format, maxCardNumberLength } = getCardTypeByValue(value) || {
    format: DEFAULT_CARD_FORMAT,
    maxCardNumberLength: 16
  }

  if (value.replace(/[^\d]/g, '').length > maxCardNumberLength) {
    return previousValue
  }

  if (format.global) {
    const match = value.match(format)
    return match ? match.join(' ') : ''
  }

  const execResult = format.exec(value.split(' ').join(''))
  if (execResult) {
    return execResult
      .splice(1, 3)
      .filter(x => x)
      .join(' ')
  }

  return value
}

export const validateCreditCard = value => {
  const cardType = getCardTypeByValue(value)

  if (!cardType) {
    return (
      <FormattedMessage
        id='formhelper.invalid_card_number'
        defaultMessage='Invalid card number'
      />
    )
  }

  if (!cardType.supported) {
    return (
      <FormattedMessage
        id='formhelper.card_type_unsupported'
        defaultMessage='Card type not supported'
      />
    )
  }
}

const CreditCardBox: React.FC<Props> = props => {
  return <TextBox {...props} />
}

type Props = { input: CommonFieldProps; meta: WrappedFieldMetaProps }

export default CreditCardBox
