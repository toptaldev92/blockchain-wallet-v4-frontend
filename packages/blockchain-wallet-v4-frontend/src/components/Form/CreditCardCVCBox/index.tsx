import { CommonFieldProps, WrappedFieldMetaProps } from 'redux-form'
import { TextBox } from 'components/Form'
import React from 'react'

export const normalizeCreditCardCVC = (value, previousValue) => {
  if (!value) return value
  if (value.length > 3) return previousValue

  const onlyNums = value.replace(/[^\d]/g, '')

  return onlyNums
}

const CreditCardCVCBox: React.FC<Props> = props => {
  return <TextBox {...props} />
}

type Props = { input: CommonFieldProps; meta: WrappedFieldMetaProps }

export default CreditCardCVCBox
