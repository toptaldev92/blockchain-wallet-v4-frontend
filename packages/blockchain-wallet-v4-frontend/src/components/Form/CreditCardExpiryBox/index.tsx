import { CommonFieldProps, WrappedFieldMetaProps } from 'redux-form'
import { FormattedMessage } from 'react-intl'
import { TextBox } from 'components/Form'
import moment from 'moment'
import React from 'react'

export const normalizeCreditCardExpiry = (value, previousValue) => {
  if (!value) return value
  if (value.length > 5) return previousValue

  const onlyNumsOrSlash = value.replace(/[^\d/]/g, '')

  return onlyNumsOrSlash
}

export const validateCreditCardExpiry = (value: string) => {
  const regex = /\d{2}\/\d{2}/

  if (!value.match(regex)) {
    return (
      <FormattedMessage
        id='formhelper.invalid_expiry_date'
        defaultMessage='Invalid Expiry Date'
      />
    )
  }

  if (Number(value.split('/')[0]) > 12) {
    return (
      <FormattedMessage
        id='formhelper.invalid_expiry_date'
        defaultMessage='Invalid Expiry Date'
      />
    )
  }

  if (moment(value, 'MM/YY') < moment()) {
    return (
      <FormattedMessage
        id='formhelper.card_expired'
        defaultMessage='Card Expired'
      />
    )
  }
}

const CreditCardExpiryBox: React.FC<Props> = props => {
  return <TextBox {...props} />
}

type Props = { input: CommonFieldProps; meta: WrappedFieldMetaProps }

export default CreditCardExpiryBox
