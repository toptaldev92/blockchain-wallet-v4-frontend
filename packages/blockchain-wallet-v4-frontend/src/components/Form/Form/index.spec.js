import { shallow } from 'enzyme'
import Form from './index.js'
import React from 'react'
import toJson from 'enzyme-to-json'

describe('Form', () => {
  it('renders correctly', () => {
    const props = { override: true, onSubmit: jest.fn() }
    const component = shallow(<Form {...props} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
