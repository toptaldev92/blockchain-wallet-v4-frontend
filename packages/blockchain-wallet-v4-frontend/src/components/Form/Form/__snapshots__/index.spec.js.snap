// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
<Form__BaseForm
  bool={false}
  override={true}
/>
`;
